packages/cashier/src/components/empty-state/empty-state.tsx

```diff
@@ -0,0 +1,23 @@
+import { Icon, Text } from '@deriv/components';
+import React from 'react';
+import './empty-state.scss';
+
+export type TEmptyStateProps = {
+    icon: string;
+    title: string | React.ReactNode;
+    description: string | React.ReactNode;
+};
+
+const EmptyState: React.FC<TEmptyStateProps> = ({ icon, title, description }) => (
```

Comment:
As this component doesn't accept the `children` prop, is it a good idea to use `React.FC` which the children prop is initiated in it?

P.S: There are some other disadvantages of using `React.FC`. Some are explained [here](https://fettblog.eu/typescript-react-why-i-dont-use-react-fc/).

Reply:
@hamid-deriv That article seems to be pretty old 👀 
We used to have `React.VFC (React.VoidFunctionComponent)` for chidless components and `React.FC (React.FunctionComponent)` for components with children. `React.VFC` was deprecated in favor of `React.FC` and `React.FC` doesn’t accept children anymore, If you want it to have children you can use `React.PropsWithChildren` generic. Something like `React.FC<React.PropsWithChildren<TEmptyStateProps>>`.

Right now if you pass a child to `EmptyState` component TS will complain about it:
<img width="1051" alt="image" src="https://user-images.githubusercontent.com/72082844/181510937-a95cb3e5-4bd6-465c-b227-29df651e4ff3.png">

Reply:
@farzin-fs I also don't like using that, it's not only about accepting children but other disadvantages(they don't affect our code here actually)

Reply:
@Bahar-FS I've removed `React.FC` and pushed the changes 🙇🏻 

Although I can see we are using it in the `appstore` package and I'm still not clear what are these disadvantages we talk about, I would be thankful if you could share a source so I can read about it 🙇🏻🙏🏻

Reply:
@farzin-fs I read about it [here](https://github.com/facebook/create-react-app/pull/8177)